crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -448,24 +491,26 @@ impl SystemNode {
 }
 
 /// Metadata for a [`Schedule`].
+#[allow(missing_docs)]
 #[derive(Default)]
 pub struct ScheduleGraph {
-    systems: Vec<SystemNode>,
-    system_conditions: Vec<Vec<BoxedCondition>>,
+    pub systems: Vec<SystemNode>,
+    pub system_conditions: Vec<Vec<BoxedCondition>>,
     system_sets: Vec<SystemSetNode>,
     system_set_conditions: Vec<Vec<BoxedCondition>>,
     system_set_ids: HashMap<InternedSystemSet, NodeId>,
     uninit: Vec<(NodeId, usize)>,
     hierarchy: Dag,
     dependency: Dag,
     ambiguous_with: UnGraphMap<NodeId, ()>,
-    ambiguous_with_all: HashSet<NodeId>,
+    pub ambiguous_with_all: HashSet<NodeId>,
     conflicting_systems: Vec<(NodeId, NodeId, Vec<ComponentId>)>,
     anonymous_sets: usize,
     changed: bool,
     settings: ScheduleBuildSettings,
-    no_sync_edges: BTreeSet<(NodeId, NodeId)>,
-    auto_sync_node_ids: HashMap<u32, NodeId>,
+
+    /// A map of [`ScheduleBuildPassObj`]es, keyed by the [`TypeId`] of the corresponding [`ScheduleBuildPass`].
+    passes: BTreeMap<TypeId, Box<dyn ScheduleBuildPassObj>>,
```

Comment:
Can we use a plain `BTreeMap` here? The order of the schedule passes surely matters once we have more complex passes. While the iteration order of a BTree is stable, I don't think it's able to be controlled in the way that we might need.

Reply:
Not sure what it means? We're already using a `BTreeMap` here.

We'd certainly want to control the ordering of those passes but I would defer that to a future PR when such a need arises.

For now, the reason we use a `BTreeMap` instead of a `Vec` here is that a `BTreeMap` makes it easier to remove passes and it guarantees unique instantiation of passes based on type, ensuring that the same type won't be accidentally added twice.

Reply:
I meant "is it correct to use a BTreeMap". I'm definitely fine to leave that until a later PR though.